docs/guide/hosting.md

```diff
@@ -0,0 +1,46 @@
+# Hosting Apostrophe in production
+
+## Server requirements
+
+We recommend hosting Apostrophe in self-hosted production environments with the following minimum specifications:
+
+| Software | Minimum Version |
+| ------------- | ------------- |
+| Ubuntu | 16+ |
+| [NGINX](https://www.nginx.com/) (or another proxy server) | Latest |
+| [Node.js](https://nodejs.org/en/) | 12.x+ |
+| [MongoDB](https://docs.mongodb.com/manual/administration/install-community/)  | 3.6+ |
+| [ImageMagick](https://imagemagick.org/script/download.php#macosx) (optional) | Any |
+
+
+| Hardware specification | Minimum recommendation |
+| ------------- | ------------- |
+| RAM | **2GB** (to support the application and database) |
+| Disk space | **20GB** (see note below) |
+
+1GB of RAM may be sufficient if using a remote database service (e.g., MongoDB Atlas). Sites running in a multicore configuration will require additional 2G or more RAM and, of course, CPU cores.
+
+::: note
+MongoDB requires a minimum of 5GB free disk space at all times. Small websites may not need much more additional space than that. We have found that 20GB is sufficient for most Apostrophe applications, but more space may be necessary to handle an especially large number of file uploads or especially large uploaded files.
+:::
+
+## Deployment basics
+
+Deployment processes will vary depending on the hosting environment, technical requirements, and team preferences. There are a few steps that all deployment processes should include.
+
+1. If Apostrophe has been running on the server already, **stop the application process(es).**
+2. **Run the data migration task: `node app @apostrophecms/migration:migrate`.** This will run both core data migrations as well as project-level migrations.
+3. **Run the build process: `node app @apostrophecms/asset:build`.** This runs automatically in development environments, but not if `NODE_ENV=production` or if the `autoBuild: false` option is set on the application. That is to ensure that the build is an intentional part of deployment and that developers can execute it along with other production build tasks as necessary.
+4. **Start (or restart) the application process(es).**
```

Comment:
This is where you stop the old instance and start the new one. No need for the site to be down during the build of the next asset generation (which is slow) and the migrations (people who have exceptions to that rule will know).